src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -786,20 +787,23 @@ static function (ChildDefinition $definition, AsPeriodicTask|AsCronTask $attribu
         }
 
         $container->registerForAutoconfiguration(CompilerPassInterface::class)
-            ->addTag('container.excluded.compiler_pass')->addTag('container.excluded')->setAbstract(true);
+            ->addTag('container.excluded', ['source' => 'because it\'s a compiler pass'])->setAbstract(true);
         $container->registerForAutoconfiguration(TestCase::class)
-            ->addTag('container.excluded.test_case')->addTag('container.excluded')->setAbstract(true);
+            ->addTag('container.excluded', ['source' => 'because it\'s a test case'])->setAbstract(true);
         $container->registerAttributeForAutoconfiguration(AsMessage::class, static function (ChildDefinition $definition) {
-            $definition->addTag('container.excluded.messenger.message')->addTag('container.excluded')->setAbstract(true);
+            $definition->addTag('container.excluded', ['source' => 'because it\'s a messenger message'])->setAbstract(true);
+        });
+        $container->registerAttributeForAutoconfiguration(Constraint::class, static function (ChildDefinition $definition) {
```

Comment:
Why would you mark every class that use the constraint attributes as excluded? 
It's the classes that extend `Constraint` that you want to exclude, isn't it?

```suggestion
        $container->registerForAutoconfiguration(Constraint::class, static function (ChildDefinition $definition) {
```

Reply:
yep, fixed